### PR TITLE
Pin all dependencies using to minor version

### PR DESCRIPTION
**Pull request by collindutter**
- [x] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

## Describe your changes
Pin all dependencies to minor versions using [caret requirements](https://python-poetry.org/docs/dependency-specification/#caret-requirements).
## Issue ticket number and link
NA

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -32,11 +32,11 @@ requests = "^2.32.0"
 cohere = { version = "^5.5.4", optional = true }
 anthropic = { version = "^0.29.0", optional = true }
 transformers = { version = "^4.39.3", optional = true, extras=["torch"] }
-huggingface-hub = { version = ">=0.13", optional = true }
+huggingface-hub = { version = "^0.24.0", optional = true }
 boto3 = { version = "^1.34.119", optional = true }
 snowflake-sqlalchemy = { version = "^1.6.1", optional = true }
 pinecone-client = { version = "^3", optional = true }
-pymongo = { version = "*", optional = true }
+pymongo = { version = "4.8.0", optional = true }
```
Comment (vasinov) [inline pyproject.toml:39]: Should we add a caret for this one as well?

**Review by vasinov — CHANGES_REQUESTED** [review]
Looks like we are pi

File: pyproject.toml (hunk 1 above)
Comment (collindutter) [reply pyproject.toml]: Yep, fixed.

**Review by dylanholmes — APPROVED** [review]
Pinning versions makes sense to me, but we should probably look for a mechanism to notify us when there is a new major version. (Maybe relying on as-needed is good enough though?)

File: griptape/drivers/prompt/amazon_sagemaker_jumpstart_prompt_driver.py [hunk 2]
```diff
@@ -113,6 +113,6 @@ def __prompt_stack_to_tokens(self, prompt_stack: PromptStack) -> list[int]:
         tokens = self.tokenizer.tokenizer.apply_chat_template(messages, add_generation_prompt=True, tokenize=True)
 
         if isinstance(tokens, list):
-            return tokens
+            return tokens  # pyright: ignore[reportReturnType] According to the [docs](https://huggingface.co/docs/transformers/main/en/internal/tokenization_utils#transformers.PreTrainedTokenizerBase.apply_chat_template), the return type is List[int].
```
[the remark this answers is not in the document]
Comment (collindutter) [reply griptape/drivers/prompt/amazon_sagemaker_jumpstart_prompt_driver.py:116]: [Yeah](https://github.com/griptape-ai/griptape/blob/16742ca713d94a533126943b9393f86eea133fb7/tests/unit/drivers/prompt/test_amazon_sagemaker_jumpstart_prompt_driver.py#L24)

**Pull request merged**